### PR TITLE
Add mention to DD_TRACE_OTEL_ENABLED

### PR DESCRIPTION
**Pull request by julianocosta89**
### What does this PR do? What is the motivation?

`DD_TRACE_OTEL_ENABLED=true`
is mentioned under https://docs.datadoghq.com/tracing/trace_collection/custom_instrumentation/java/otel/#setup and should also be called out under https://docs.datadoghq.com/tracing/trace_collection/library_config/java.

### Merge instructions

- [x] Please merge after reviewing

### CODE REVIEW

**Comment by mcculls** [comment]
We should remove this from the experimental section at the same time, because `DD_TRACE_OTEL_ENABLED` replaces it:
```
dd.integration.opentelemetry.experimental.enabled : Environment Variable: DD_INTEGRATION_OPENTELEMETRY_EXPERIMENTAL_ENABLED
Default: false
Enables experimental OpenTelemetry features. When you enable this setting, you can use the OpenTelemetry tracing API backed by the Datadog Trace Agent.
Available since version 1.10.0
```

**Review by PerfectSlayer — APPROVED** [review]

File: content/en/tracing/trace_collection/library_config/java.md [hunk 1]
```diff
@@ -317,6 +317,11 @@ When `true`, the tracer generates 128 bit Trace IDs, and encodes Trace IDs as 32
 **Default**: `false`<br>
 When `true`, the tracer will inject 128 bit Trace IDs as 32 lowercase hexadecimal characters with zero padding, and 64 bit Trace IDs as decimal numbers. Otherwise, the tracer always injects Trace IDs as decimal numbers.
 
+`dd.trace.otel.enabled`
+: **Environment Variable**: `DD_TRACE_OTEL_ENABLED`<br>
+**Default**: `false`<br>
+When `true`, enables OpenTelemetry based tracing for [custom][16] instrumentation.
```
Comment (rtrieu) [inline content/en/tracing/trace_collection/library_config/java.md:323]:
```suggestion
When `true`, OpenTelemetry-based tracing for [custom][16] instrumentation is enabled.
```

**Review by rtrieu — CHANGES_REQUESTED** [review]
Minor suggestion but otherwise looks good. Thanks!

**Review by rtrieu — APPROVED** [review]
Looks good!

**Pull request merged**